src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1836,9 +1835,16 @@ private static unsafe void DecodeObjects(object?[] decodedObjects, Type[] parame
                 }
 
             BytePtr:
-                var blob = new byte[data->Size];
-                Marshal.Copy(dataPointer, blob, 0, blob.Length);
-                decoded = blob;
+                if (data->Size == 0)
+                {
+                    decoded = Array.Empty<byte>();
+                }
+                else
+                {
+                    var blob = new byte[data->Size];
+                    Marshal.Copy(data->DataPointer, blob, 0, blob.Length);
+                    decoded = blob;
+                }
```

Comment:
This would be simpler and safer as:
```C#
decoded = new Span<byte>(data->DataPointer, data->Size).ToArray();
```

Reply:
That is what I had used initially, but it sadly involves duplicating the logic under `#if #else` as `EventSource` is also being built standalone

Reply:
Why not reference the netstandard span package from the downlevel solution?

Either way,  I don't think we should penalize the latest platform because of the source sharing.

Reply:
@brianrob, we still maintain/ship the separate EventSource package? Who uses that?

Reply:
> Either way, I don't think we should penalize the latest platform because of the source sharing.

100% agree with this.

> Who uses that?

This periodically comes up (see: https://github.com/dotnet/runtime/pull/32276#issuecomment-587094961). This is building [this package](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.EventSource.Redist/6.0.0-preview.4.21253.7), which is used to ship EventSource features out-of-band for older framework releases. Download counts have been dwindling since 2015. The 5.0 version has ~320k downloads in 7 months.

Reply:
> which is used to ship EventSource features out-of-band for older framework releases

Are we still adding new EventSource features it's important to be able to release in this vehicle?  Can we just stop shipping new versions of it?

Reply:
_I'm_ okay with moving that to a servicing life cycle. Brian has a better sense of what impact that would have though, so I would defer to his input before ripping it out.

CC @noahfalk

Reply:
> a relief valve to be able to fix issues on .NET Framework

To make sure I understand, this doesn't actually unify with what's in .NET Framework, right?  Someone has to explicitly switch from using the EventSource in .NET Framework to this differently namespaced type in the OOB?

Reply:
Yes, that's right - it does not unify.  Someone will have to explicitly consume the NuGet package.  The namespace is also different - `Microsoft.Diagnostics.Tracing` instead of `System.Diagnostics.Tracing`.